implementation/jwt-build/src/test/java/io/smallrye/jwt/build/JwtSignTest.java

```diff
@@ -593,19 +593,36 @@ private void doTestSignedExistingClaims(String jwt) throws Exception {
 
     @Test
     void signClaimsEllipticCurve() throws Exception {
-        EllipticCurveJsonWebKey jwk = createECJwk();
+        EllipticCurveJsonWebKey ecJwk = createECJwk();
 
         String jwt = Jwt.claims()
                 .claim("customClaim", "custom-value")
-                .sign(jwk.getEcPrivateKey());
+                .claim("evidence", ecJwk.getECPublicKey())
+                .jws().jwk(ecJwk.getECPublicKey())
+                .sign(ecJwk.getEcPrivateKey());
```

Comment:
Only asking - should there be any validation that a `jwk` corresponds to the signing key?

Reply:
Hi @MikeEdgar No, I'm just using the same key pair for the test convenience. The signature verification, in the smallrye-jwt at least will not depend on this header, it will expect the verification keys configured or fetched from the OIDC provider.
I decided to this PR to support various DPoP test cases, to create proof tokens as in https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop-16#name-dpop-proof-jwt-syntax, etc...